docs/class-parser.md

```diff
@@ -0,0 +1,34 @@
+Class: Parser
+=============
+
+Parser is the class that will translate strings (starting with yamal) into properties (json) between storage and services
+
+## Interface
+
+### Class Attribute: `properties`
+
+```
+Parser.properties
+```
+
+The unmodified properties that need to be translated from human readable to json for services to read.
+
+### Class Attribute: `sources`
+
+```
+Parser.sources
+```
+
+The source of the properties. ie S3, Conqueso, ect
+
+### Constructor(properties (buffer), plugin_type (string))
```

Comment:
To me, this raises the question as to whether we want a monolithic parser that parses all expected input or if we want discrete parsers for every data source. Per a conversation with @fmitchell-r7 I'm leaning more towards the latter. It fits a little better into node's paradigm of using small, discrete packages and if we get to a point where we're duplicating code, we can think about extracting it into separate helper packages.